source/math/BoolVectorList.ooc

```diff
@@ -0,0 +1,119 @@
+	init: func ~default {
+		this super()
+	}
+	init: func ~heap (capacity: Int){
+		super(capacity)
+	}
+	init: func ~fromVectorList (other: VectorList<Bool>) {
+		this super(other _vector)
+		this _count = other count
+	}
+	totalTrues: func -> Int {
+		result := 0
+		for (i in 0..this count)
+			result += (this[i] ? 1 : 0)
+		result
+	}
+	totalFalses: func -> Int {
+		this _count - totalTrues()
+	}
+	operator [] <T> (index: Int) -> T {
+		this as VectorList<Bool> _vector[index]
+	}
+	operator []= (index: Int, item: Bool) {
+		this _vector[index] = item
+	}
+	/*operator && (other: This) -> This {
+		minimumCount := this count < other count ? this count : other count
+		result := This new(minimumCount)
+		for (i in 0..minimumCount)
+			result add(this[i] && other[i])
+		result
+	}
+	operator || (other: This) -> This {
+		minimumCount := this count < other count ? this count : other count
+		result := This new(minimumCount)
+		for (i in 0..minimumCount)
+			result add(this[i] || other[i])
+		result
+	}*/
+	reverse: func -> This {
+		result := This new(this _count)
+		for (i in 1..(this _count + 1))
+			result add(this[this _count - i])
+		result
+	}
+	erosion: func (elementSize: Int) -> This {
+		result := This new(this _count)
+		halfSize := round((elementSize - 1) / 2) as Int
+		currentSequence := This new()
+		for (i in 0..this _count) {
+			currentSequence = this getSlice((i - halfSize) clamp(0, this count - 1), (i + halfSize) clamp(0, this count - 1))
+			result add(currentSequence totalTrues() == currentSequence count)
+		}
+		result
+	}
+	dilation: func (elementSize: Int) -> This {
+		result := This new(this _count)
+		halfSize := round((elementSize - 1) / 2) as Int
+		currentSequence := This new()
+		for (i in 0..this _count) {
+			currentSequence = this getSlice((i - halfSize) clamp(0, this count - 1), (i + halfSize) clamp(0, this count - 1))
+			result add(currentSequence totalTrues() > 0)
+		}
+		result
+	}
+	opening: func (elementSize: Int) -> This {
+		erosion(elementSize) dilation(elementSize)
+	}
+	closing: func (elementSize: Int) -> This {
+		dilation(elementSize) erosion(elementSize)
+	}
+	toFloatVectorList: func (floatForFalse := 0.0f, floatForTrue := 1.0f) -> FloatVectorList {
+			result := FloatVectorList new(this _count)
+		for (i in 0..this _count)
+			result add(this[i] ? floatForTrue : floatForFalse)
+		result
+	}
+	toString: func() -> String {
+		result := ""
+		for (i in 0..this _count)
+			result = result >> this[i] toString() >> "\n"
+		result
+	}
+	or: func (other: This) -> This {
+		minimumCount := this count < other count ? this count : other count
+		result := This new(minimumCount)
+		for (i in 0..minimumCount)
+			result add(this[i] || other[i])
+		result
+	}
+	and: func (other: This) -> This {
+		minimumCount := this count < other count ? this count : other count
+		result := This new(minimumCount)
+		for (i in 0..minimumCount)
+			result add(this[i] && other[i])
+		result
+	}
+}
```

Comment:
Check this file for style violations with magic: https://github.com/cogneco/magic